### ISSUE TITLE
convertValuetoArray($value) not compatible with __call()

### ISSUE THREAD

**Issue by ametad** [PHPExpertsInc/SimpleDTO#19]
Hi @hopeseekr ,

Thank you for your work, this package. I am using this in favor of the Spatie package because I also think DTOs should be immutable. 

I run into a problem with the `$dto->toArray()` method when I have a `\Illuminate\Http\UploadedFile` in the DTO. Because it has magic function `__call()` implemented (by the  `\Illuminate\Support\Traits\Macroable` trait), the next check always returns true (even when no `toArray` exists). Therefore I get this error: 

> Method Illuminate\Http\UploadedFile::toArray does not exist.

The problem is known: https://www.php.net/manual/en/function.is-callable.php#118623

In `\PHPExperts\SimpleDTO\SimpleDTO::convertValueToArray` there is this check for every data attribute value:

```php
        if (is_object($value)) {
            if (is_callable([$value, 'toArray']) && !($value instanceof Carbon)) {
                return $value->toArray();
            }

            if ($value instanceof \stdClass) {
                return (array) $value;
            }
        }
```

source: https://github.com/PHPExpertsInc/SimpleDTO/blob/f7b027eae78d16f8042b12ff4942e61e7321cd8f/src/SimpleDTO.php#L328

The problem: `is_callable` has trouble figuring out that `UploadedFile` has no callable `toArray`, it says `true` but instead it is `false`.

I think the following would fix the problem:

```php
        if (is_object($value)) {
            if (method_exists($value, 'toArray') && !($value instanceof Carbon)) {   // <=== this would work
                return $value->toArray();
            }

            if ($value instanceof \stdClass) {
                return (array) $value;
            }
        }
```

Would you consider implementing this fix?

**Comment by hopeseekr** [comment]
Sure. I'll test it later tonight.

**Comment by hopeseekr** [comment]
Available in version 2.6.0!

**Comment by ametad** [comment]
Thank you.
